cartography/intel/aws/iam.py

```diff
@@ -270,70 +294,66 @@ def load_group_memberships(neo4j_session, group_memberships, aws_update_tag):
             )
 
 
-def _find_roles_assumable_in_policy(policy_data):
-    ret = []
-    statements = policy_data["PolicyDocument"]["Statement"]
-    if isinstance(statements, dict):
-        statements = [statements]
-    for statement in statements:
-        parsed_statement = policyuniverse.statement.Statement(statement)
-        if parsed_statement.effect == 'Allow' and 'sts:assumerole' in parsed_statement.actions_expanded:
-            ret.extend(list(parsed_statement.resources))
-    return ret
-
-
 @timeit
-def load_group_policies(neo4j_session, group_policies, aws_update_tag):
-    ingest_policies_assume_role = """
-    MATCH (group:AWSGroup{arn: {GroupArn}})
-    WITH group
-    MERGE (role:AWSRole{arn: {RoleArn}})
-    ON CREATE SET role.firstseen = timestamp()
-    SET role.lastupdated = {aws_update_tag}
-    WITH role, group
-    MERGE (group)-[r:STS_ASSUMEROLE_ALLOW]->(role)
-    ON CREATE SET r.firstseen = timestamp()
-    SET r.lastupdated = {aws_update_tag}
+def get_policies_for_principal(neo4j_session, principal_arn):
```

Comment:
If possible we should avoid reading from the graph and then processing it through python because of the IO overhead from transferring stuff between Neo4j and python. Instead, we should try to keep all of this logic in Neo4j itself by merging on policy IDs.

Reply:
When I profile this, the overhead is bearly measurable even in the full graph. This is a clone of exiting behavior that existed pre refactor.

Reply:
Fair point, I won't block on this. In future let's avoid doing this unless there's no way around it; there's lots of messy practices in the code :\